### DISCUSSION TITLE
Efficient way to determine height/Y on any given X/Z position of a terrain?

### DISCUSSION THREAD

**Discussion by Deltt** [TokisanGames/Terrain3D#345] [Questions] [answered]
Hello there!

First of all, thank you for providing this incredible tool, its been highly beneficial for our project.

We are currently working on a squad based RTS (meaning that not every individual unit is pathfinding, instead they are commanded in fixed groups and always stick around the origin position of their group). Being an RTS, we try to safe as much resources as possible to get a high amount of units/squads and their behaviors going.

I'm looking for guidance if the tool also provides or enables an **efficient** way to determine the Y position of individual units on a given X,Z coordinate, as we are trying to avoid any and all physics. Up until now we handled it with raycasts for each individual unit, which in our tests wasn't sufficiently scalable.

I'd greatly appreciate it if anyone can point me towards something helpful!

Happy eastern,

Deltt

**Comment by TokisanGames** [comment, marked as answer]
This is answered at length in the docs. You'll find multiple options in the API, and the integrating with Terrain3D document.

**Reply by Deltt** [reply]
Thank you, I don't know how I missed that!

Looking at it, I think there might be a typo in the "Query the height at any position" section:
`fox x in img.get_width():`

**Reply by TokisanGames** [reply]
Thanks. Fixed in 501f043fbbdd2586a0dbef1c16fbbaf5e0fdc8d0
